src/libraries/Microsoft.Extensions.Hosting/src/Internal/ApplicationLifetime.cs

```diff
@@ -107,17 +107,5 @@ public void NotifyStopped()
                                          ex);
             }
         }
-
-        private static void ExecuteHandlers(CancellationTokenSource cancel)
-        {
-            // Noop if this is already cancelled
-            if (cancel.IsCancellationRequested)
```

Comment:
What is the reason to no longer include this "noop" case?

Reply:
`CancellationTokenSource` has the exact same check already, so calling it multiple times has no effect.

Reply:
OK, I verified this is correct for both Core and .NET Framework.